hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -1010,47 +1012,49 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 
 	// Set the ignition server availability condition by checking its deployment.
 	{
-		// Assume the server is unavailable unless proven otherwise.
-		newCondition := metav1.Condition{
-			Type:   string(hyperv1.IgnitionEndpointAvailable),
-			Status: metav1.ConditionUnknown,
-			Reason: hyperv1.StatusUnknownReason,
-		}
-		// Check to ensure the deployment exists and is available.
-		deployment := ignitionserver.Deployment(controlPlaneNamespace.Name)
-		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(deployment), deployment); err != nil {
-			if apierrors.IsNotFound(err) {
+		if _, exists := hcp.Annotations[hyperv1.DisableIgnitionServerAnnotation]; !exists {
+			// Assume the server is unavailable unless proven otherwise.
+			newCondition := metav1.Condition{
+				Type:   string(hyperv1.IgnitionEndpointAvailable),
+				Status: metav1.ConditionUnknown,
+				Reason: hyperv1.StatusUnknownReason,
+			}
+			// Check to ensure the deployment exists and is available.
+			deployment := ignitionserver.Deployment(controlPlaneNamespace.Name)
+			if err := r.Client.Get(ctx, client.ObjectKeyFromObject(deployment), deployment); err != nil {
+				if apierrors.IsNotFound(err) {
+					newCondition = metav1.Condition{
+						Type:    string(hyperv1.IgnitionEndpointAvailable),
```

Comment:
without the check this condition would legitimately signal this state here, are you intentionally wanting to avoid that?

Reply:
Good point--if the state is purely informational, I think it's fine to avoid it being set, but now I'm thinking about it, I'm worried if that state is referenced elsewhere or if it's used to determine the state of the HostedCluster resource. I'm not certain of that yet--do you happen to know?